go/protocol/stellar1/extras.go

```diff
@@ -66,6 +67,39 @@ func (t TransactionID) Eq(b TransactionID) bool {
 	return t == b
 }
 
+type byTxID []TransactionID
+
+func (s byTxID) Len() int {
+	return len(s)
+}
+func (s byTxID) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s byTxID) Less(i, j int) bool {
+	return s[i] < s[j]
+}
+
+func TransactionIDsPtrEq(x, y *[]TransactionID) bool {
```

Comment:
args need to be a pointer to a slice and not just a slice value because we need to handle null txIDs in the client header, which makes the logic in general a bit more complicated